### ISSUE TITLE
Support Angular CLI

### ISSUE THREAD

**Issue by birkskyum** [oven-sh/bun#4060]
Angular CLI - [Website](https://angular.io/cli) - [GitHub](https://github.com/angular/angular-cli)

### What version of Bun is running?

0.7.3

### What platform is your computer?

Darwin 22.5.0 arm64 arm

### What steps can reproduce the bug?

Execute the following
```
bun --bun x ng --help
```

### What is the expected behavior?

The below works if `--bun` is removed

```zig

➜ bun --bun x ng --help      
ng <command>

Commands:
  ng add <collection>            Adds support for an external library to your project.
  ng analytics                   Configures the gathering of Angular CLI usage metrics.
  ng build [project]             Compiles an Angular application or library into an output directory named dist/ at the given output path.                                                            [aliases: b]
  ng cache                       Configure persistent disk cache and retrieve cache statistics.
  ng completion                  Set up Angular CLI autocompletion for your terminal.
  ng config [json-path] [value]  Retrieves or sets Angular configuration values in the angular.json file for the workspace.
  ng deploy [project]            Invokes the deploy builder for a specified project or for the default project in the workspace.
  ng doc <keyword>               Opens the official Angular documentation (angular.io) in a browser, and searches for a given keyword.                                                                [aliases: d]
  ng e2e [project]               Builds and serves an Angular application, then runs end-to-end tests.                                                                                                [aliases: e]
  ng extract-i18n [project]      Extracts i18n messages from source code.
  ng generate                    Generates and/or modifies files based on a schematic.                                                                                                                [aliases: g]
  ng lint [project]              Runs linting tools on Angular application code in a given project folder.
  ng new [name]                  Creates a new Angular workspace.                                                                                                                                     [aliases: n]
  ng run <target>                Runs an Architect target with an optional custom builder configuration defined in your project.
  ng serve [project]             Builds and serves your application, rebuilding on file changes.                                                                                                      [aliases: s]
  ng test [project]              Runs unit tests in a project.                                                                                                                                        [aliases: t]
  ng update [packages..]         Updates your workspace and its dependencies. See https://update.angular.io/.
  ng version                     Outputs Angular CLI version.                                                                                                                                         [aliases: v]

Options:
  --help  Shows a help message for this command in the console.                                                                                                                                          [boolean]

For more information, see https://angular.io/cli/.


```

### What do you see instead?

```sh
55 |     if (process.stdout instanceof tty_1.WriteStream) {
           ^
TypeError: instanceof called on an object with an invalid prototype property.
      at supportColor (/Users/admin/repos/analog-project/node_modules/@angular/cli/src/utilities/color.js:55:8)
```

### Additional information

Blocked by:
- #4061 
- #2081 
- https://github.com/oven-sh/bun/pull/4179

**Comment by sp90** [comment]
Tracking of this feature is currently living here: #2081

**Comment by birkskyum** [comment]
Closed by #4179
